### ISSUE TITLE
Read Excel file take too much time

### ISSUE THREAD

**Issue by ravishankersingh19** [dilshod/xlsx2csv#201]
I really appreciate your library it convert my .**XLSX** file in 8 minutes. Need your help for reading excel file, clean and upload in MySQL.

My XLSX size : 400 MB and
Format: 

3 | 29 | 65 | 12 | 34 | 24 |   | 3 | 17 | 55 | 36 | 68 | 25 |   | 33 | 7 | 57 | 40 | 52 | 1 |   | 47 | 21 | 61 | 24 | 32 | 2 |   | 59 | 7 | 47 | 18 | 64 | 3 |   | 3 | 21 | 61 | 38 | 68 | 4 |   | 69 | 31 | 37 | 2 | 56 | 5 |   | 17 | 57 | 67 | 40 | 62 | 6 |   | 27 | 37 | 1 | 13 | 16 | 7 |   | 47 | 57 | 5 | 29 | 54 | 8 |   | 41 | 51 | 3 | 53 | 26 | 9 |   | 25 | 63 | 67 | 69 | 30 | 10 |   | 46 | 50 | 64 | 25 | 41 | 11 |   | 2 | 18 | 54 | 49 | 53 | 12 |   | 4 | 12 | 70 | 19 | 49 | 13 |   | 26 | 60 | 70 | 17 | 31 | 14 |   | 38 | 16 | 56 | 13 | 39 | 15 |   | 46 | 50 | 70 | 11 | 51 | 16 |   | 58 | 16 | 46 | 65 | 67 | 17 |   | 28 | 60 | 64 | 19 | 37 | 18 |   | 14 | 48 | 52 | 60 | 45 | 19 |   | 24 | 54 | 20 | 36 | 33 | 20 |   | 20 | 26 | 44 | 62 | 3 | 21 |   | 48 | 68 | 50 | 54 | 41 | 22 |   | 22 | 26 | 44 | 46 | 70 | 23 |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

Basically 

I am try with openpyxl library to reading this file but its getting too much could you please provided you way to read this file more fast.
```
 from openpyxl import load_workbook
from openpyxl.utils.dataframe import dataframe_to_rows


start_time = datetime.datetime.now()

wb = load_workbook(filename='Book1.xlsx', read_only=True, data_only=True)
ws=wb.active
column = ws.max_column
row = ws.max_row

for i in range(1, row+1):
    for j in range(1, column+1):
        cell_obj = ws.cell(row=i, column=j)
        if cell_obj.value != None:
            print(cell_obj.value)
    print('----------------This Row this execute--------------------------------')

end_time = datetime.datetime.now()
print(end_time-start_time)
```

**Comment by Nabakamal** [comment]
Can you try using the dask library? If you have nodes (read: servers/other machines) that can do "work" on behalf of your "scheduling" machine then it can be a good fit.  URL is: https://distributed.dask.org/en/latest/. You can pass on the creation of each sheet to a "worker". Please look it up and see if that helps you.

**Comment by ravishankersingh19** [comment]
Its really good suggestion. 
I have done its divide-and-conquer approach. will try dask library also for performance analysis.

Thanks for your comment !!!

**Comment by radoeka** [comment]
Issue seems to be answered.
Can this issue be closed?  If so, close it, please.

**Comment by ravishankersingh19** [comment]
I have done by another approach. thank you @Nabakama @xlsx2csv
